Add bulk:deleteByQuery (#1614)
kuzzleio/kuzzle: doc/2/api/controllers/document/delete-by-query/index.md

The request fails if the number of documents returned by the search query exceeds the `documentsWriteCount` server configuration (see the [Configuring Kuzzle](/core/2/guides/essentials/configuration) guide).

> This behavior is to prevent Kuzzle from Out of Memory error or DOS because every deleted document has to be fetched first to trigger realtime notifications.

Reviewer: I don't think using `deleteByQuery` can be considered a DoS attack.

```suggestion
This behavior aims at limiting the pressure on memory and on real-time notifications.
```

Author: I meant that the OOM can lead to DOS but I agree it could be confusing

- `collection`: collection name
- `index`: index name

> ### Optional:

Reviewer: ```suggestion
### Optional
```

> - `refresh`: if set to `wait_for`, Kuzzle will not respond until the created/replaced documents are indexed

Reviewer: ```suggestion
- `refresh`: if set to `wait_for`, Kuzzle will not respond until the deleted documents are removed from the search indexes
```
